pytrip/tripexecuter2/execute.py

```diff
@@ -0,0 +1,496 @@
+
+        return out
+
+    def execute(self, plan, _callback=None):
+        """
+        Executes the Plan() object using TRiP98.
+        """
+        logger.debug("Execute TRiP98...")
+        self._callback = _callback  # TODO: check if GUI really needs this.
+        self._pre_execute(plan)  # prepare directory where all will be run, put files in it.
+        plan.save_exec(plan._exec_path)  # add the .exec as well
+        self._run_trip(plan)  # run TRiP
+        self._finish(plan)
+
+    def _pre_execute(self, plan):
+        """
+        Prepare a temporary working directory where TRiP will be executed.
+        Sets:
+        _temp_dir where all will be executed
+        _exec_path generates full path to the file name.exec file will be stored
+        """
+
+        # attach working dir to plan, but expanded for environment variables
+        plan._working_dir = os.path.expandvars(plan.working_dir)
+
+        # We will not run TRiP98 in working dir, but in an isolated subdirectory which will be
+        # uniquely created for this purpose, located after the working dir.
+        # This contents of this subdir will be termed "the package", as it is a clean environment.
+        # The name of the subdir will be stored in plan._temp_dir
+        # It may be safely deleted afterwards, once the resulting files are copied back to the plan.working_dir
+
+        if not hasattr(plan, "_temp_dir"):
+            import tempfile
+            plan._temp_dir = tempfile.mkdtemp(prefix='trip98_', dir=plan._working_dir)
+
+        plan._temp_dir = os.path.join(plan._working_dir,
+                                      plan._temp_dir)
+
+        plan._exec_path = os.path.join(plan._temp_dir,
+                                       plan.basename + ".exec")
+
+        logger.debug("Created temporary working directory {:s}".format(plan._temp_dir))
+
+        _flist = []  # list of files which must be copied to the package.
+
+        if plan.incube_basename:
+            _flist.append(os.path.join(plan._working_dir, plan.incube_basename + ".dos"))
+            _flist.append(os.path.join(plan._working_dir, plan.incube_basename + ".hed"))
+
+        for _field in plan.fields:
+            if _field.use_raster_file:
+                _flist.append(os.path.join(plan._working_dir, _field.basename + ".rst"))
+
+        # once the file list _flist is complete, copy it to the package location
+        for _fn in _flist:
+            logger.debug("Copy {:s} to {:s}".format(_fn, plan._temp_dir))
+            shutil.copy(_fn, plan._temp_dir)
+
+        # Ctx and Vdx files are not copied, but written from the objects passed to Execute() during __init__.
+        # This gives the user better control over what Ctx and Vdx should be based for the planning.
+        self.ctx.write(os.path.join(plan._temp_dir, self.ctx.basename + ".ctx"))  # will also make the hed
+        self.vdx.write(os.path.join(plan._temp_dir, self.vdx.basename + ".vdx"))
```

Comment:
I am not sure how TRiP98 will behave when providing sporadic full paths or relative files (and even worse where they will output them). So in order to avoid any troubles here (and to be more time efficient) I accepted this data duplication (which is on disk and not in mem), and instead have a clean and simple environment where I know what is going on.